File: python/selenium/configs.py
kobitonServerUrl = 'https://userName:apiKey@api.kobiton.com/wd/hub'

# 100 seconds
session_timeout = 120

desired_caps_android_web = {
  'sessionName': '[Python] Android web',
  'sessionDescription': 'This is an example for Android web testing',
  'deviceOrientation':  'portrait',
  'browserName':        'chrome',
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         'Galaxy S5',
  'platformName':       'Android',
  'newCommandTimeout':  120
}

desired_caps_android_app = {
  'sessionName': '[Python] Android app',
  'sessionDescription': 'This is an example for Android app testing',
  'deviceOrientation':  'portrait',
  'app':                'https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.apk',
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         'Galaxy S5',
  'platformName':       'Android',
  'newCommandTimeout':  120
}

desired_caps_ios_web = {
  'sessionName': '[Python] iOS web',
  'sessionDescription': 'This is an example for iOS web testing',
  'deviceOrientation':  'portrait',
  'browserName':        'safari', 
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         'iPhone 7',
  'platformName':       'iOS',
  'newCommandTimeout':  120
}

desired_caps_ios_app = {
  'sessionName': '[Python] iOS app',
  'sessionDescription': 'This is an example for iOS app testing',
  'deviceOrientation':  'portrait',
  'app':                'https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.ipa',
  'captureScreenshots': True,
  'deviceGroup':        'KOBITON',
  'deviceName':         'iPhone 7',
  'platformName':       'iOS',
  'newCommandTimeout':  120
}

File: ruby_rspec/configs.rb
class DesiredCapabilities
  USER_NAME = ''
  API_KEY = ''
  
  def desired_caps_android_web
    desired_caps = {
      :caps => {
        :sessionName        => '[Rspec] Android Web',
        :sessionDescription => 'This is an example for Android Web testing',
        :deviceOrientation  => 'portrait',
        :browserName        => 'chrome',
        :captureScreenshots => true,
        :deviceGroup        => 'KOBITON',
        :deviceName         => 'Galaxy*',
        :platformName       => 'Android',
        :newCommandTimeout  => 300
      },
      :url => 'https://#{USER_NAME}:#{API_KEY}@api.kobiton.com/wd/hub'
    }
    return desired_caps
  end    

  def desired_caps_android_app
    desired_caps = {
      :caps => {
        :sessionName        => '[Rspec] Android App',
        :sessionDescription => 'This is an example for Android app testing', 
        :deviceOrientation  => 'portrait', 
        :app                => 'https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.apk', 
        :captureScreenshots => true, 
        :deviceGroup        => 'KOBITON',
        :deviceName         => 'Galaxy*',
        :platformName       => 'Android'
      },
      :url => 'https://#{USER_NAME}:#{API_KEY}@api.kobiton.com/wd/hub'
    }
    return desired_caps
  end

  def desired_caps_ios_web
    desired_caps = {
      :caps => {
        :sessionName        => '[Rspec] iOS Web',
        :sessionDescription => 'This is an example for iOS Web testing',
        :deviceOrientation  => 'portrait',
        :browserName        => 'safari',
        :captureScreenshots => true,
        :deviceGroup        => 'KOBITON',
        :deviceName         => 'iPhone*',
        :platformName       => 'iOS'
      },
      :url => 'https://#{USER_NAME}:#{API_KEY}@api.kobiton.com/wd/hub'
    }
    return desired_caps
  end

  def desired_caps_ios_app
    desired_caps = {
      :caps => {
        :sessionName        => '[Rspec] iOS App',
        :sessionDescription => 'This is an example for iOS app testing', 
        :deviceOrientation  => 'portrait', 
        :app                => 'https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.ipa', 
        :captureScreenshots => true, 
        :deviceGroup        => 'KOBITON',
        :deviceName         => 'iPhone*',
        :platformName       => 'iOS'
      },
      :url => 'https://#{USER_NAME}:#{API_KEY}@api.kobiton.com/wd/hub'
    }
    return desired_caps
  end
end
